### DISCUSSION TITLE
Can't resolve '@/util' in when compiling external package

### DISCUSSION THREAD

**Discussion by samdharris** [laravel/nova-issues#4572] [Q&A] [answered]
- Laravel Version: 9.19.0
- Nova Version: 4.12.2
- PHP Version: 8.0.13
- Operating System and Version: Mac OS 12.4
- Browser type and version: Chrome 103.0.5060.114
- Reproduction Repository: https://github.com/Natterhub-Ltd/nova-fields
- WIP Upgrade Commit: https://github.com/Natterhub-Ltd/nova-fields/commit/2405b8ea4bdd52ffa3ed190efa0b65bb54c52df0

### Description:
When running the `npm run prod` command on the package above, I get the below screenshot:

<img width="1112" alt="Screenshot 2022-07-13 at 15 36 34" src="https://user-images.githubusercontent.com/8377862/178760419-431f1565-7d37-46e6-9f88-685d989f7b73.png">

This package was originally a Nova 3 package which I'm upgrading to Nova 4 to suit our business needs so I'm not entirely sure what I've missed as part of the upgrade process. Any ideas?

### Detailed steps to reproduce the issue on a fresh Nova installation:
1. clone provided repository
2. run `composer install && npm ci && npm run nova:install`
3. within `vendor/laravel/nova`
    1. run `mv webpack.mix.js.dist webpack.mix.js`
    2. run `npm run dev`
4. within the root of the cloned repo, run `npm run prod`

**Edit 1:** Added link to commit of initial upgrade work for easier viewing

**Comment by samdharris** [comment, marked as answer]
Update: I have narrowed the issue to be whenever you include a mixin from the Nova source that used to be in the `laravel-nova` npm package (such as InteractsWithDates).

**Reply by crynobone** [reply]
The public API for 3rd party packages are only mixins defined under `resources/js/mixins/packages.js`. You need to resolve `@/utils` on your own if you need this.

**Reply by samdharris** [reply]
Okay dokkie, thanks @crynobone
